lib/manageiq/rpm_build/build_uploader.rb

```diff
@@ -0,0 +1,59 @@
+module ManageIQ
+  module RPMBuild
+    class BuildUploader
+      attr_reader :release
+
+      def initialize(options = {})
+        @release = options[:release]
+      end
+
+      def upload
+        return unless OPTIONS.rpm_repository.s3_api.access_key && OPTIONS.rpm_repository.s3_api.secret_key
+
+        require 'digest'
+        puts "Uploading files..."
+        BUILD_DIR.glob(File.join('**', '*.rpm')).each do |file|
+          product_directory = release ? OPTIONS[:product_name] : "#{OPTIONS[:product_name]}-nightly"
+          destination_name = File.join("builds", product_directory, File.basename(file))
+          md5sum = Digest::MD5.file(file).hexdigest
+
+          if md5sum == remote_etag(destination_name)
+            puts "  Skipping existing file: #{destination_name}"
+          else
+            puts "  Uploading: #{destination_name}"
+            File.open(file, 'rb') do |content|
+              client.put_object(
+                :bucket => OPTIONS.rpm_repository.s3_api.bucket,
+                :key    => destination_name,
+                :body   => content,
+                :acl    => 'public-read'
+              )
```

Comment:
Alterantively,

```ruby
client.put_object(
  :bucket => OPTIONS.rpm_repository.s3_api.bucket,
  :key    => destination_name,
  :body   => File.read(file, :mode => "rb"),
  :acl    => 'public-read'
)
```

Reply:
Won't `File.read` read the whole file into memory?  Some of these RPMs can be quite large.  This was based on the documentation for [streaming a file from disk](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html#put_object-instance_method)

Reply:
Ah I didn't realize it would stream via the IO.  If so, then yeah, this way is better.